jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/TransferUtils.kt

```diff
@@ -75,18 +75,22 @@ fun S3Client.download(
     destination: Path,
     message: String = message("s3.download.object.progress", key),
     startInBackground: Boolean = true
-): CompletionStage<GetObjectResponse> = download(project, bucket, key, destination.outputStream(), message, startInBackground)
+): CompletionStage<GetObjectResponse> = download(
+    project,
+    GetObjectRequest.builder().bucket(bucket).key(key).build(),
+    destination.outputStream(),
+    message,
+    startInBackground
+)
 
 fun S3Client.download(
     project: Project,
-    bucket: String,
-    key: String,
+    request: GetObjectRequest,
```

Comment:
i'd rather this take an nullable versionId parameter instead of passing the client request here

Reply:
Done 👍